### ISSUE TITLE
Webkit on Windows does not support page.wait_for_load_state('networkidle')

### ISSUE THREAD

**Issue by Pomax** [microsoft/playwright-python#659]
I'm running the following fairly trivial code (with playwright installed in a venv, using `pip install playwright` followed by `python -m playwright install` to make sure all three browsers are available):

```python
import sys
import argparse
import asyncio
from playwright.async_api import async_playwright

parser = argparse.ArgumentParser(description='Take a screenshot of a web page.')
parser.add_argument('url', help='The URL for the web page.')
parser.add_argument('-f', '--file', default='screenshot.png', help='The file path to save the screenshot to, defaults to "screenshot.png".')
parser.add_argument('-w', '--width', type=int, default='1200', help='The browser width in pixels, defaults to 1200.')
args = parser.parse_args()


async def main():
    async with async_playwright() as p:
        for browser_type in [p.chromium, p.firefox, p.webkit]:
            browser = await browser_type.launch(
                headless=True,
                args=[f'--window-size={args.width},800']
            )

            page = await browser.new_page()
            await page.goto(args.url)
            await page.wait_for_load_state('networkidle')
            await page.wait_for_timeout(2000)

            document_height = await page.evaluate("""
                () => {
                    const { body } = document;
                    const html = document.querySelector(`html`);
                    return Math.max(
                        body.scrollHeight,
                        body.offsetHeight,
                        html.clientHeight,
                        html.scrollHeight,
                        html.offsetHeight
                    );
                }
            """)

            await page.set_viewport_size({
                'width': args.width,
                'height': document_height,
            })

            await page.screenshot(path=f'example-{browser_type.name}.png')
            await browser.close()

asyncio.run(main())
```

I invoke this code with a simple `python test.py https://stackoverflow.com/questions`, and while chromium and firefox work fine, webkit just dies:

```
Traceback (most recent call last):
  File "test.py", line 56, in <module>
    asyncio.run(main())
  File "c:\program files\python37\lib\asyncio\runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "c:\program files\python37\lib\asyncio\base_events.py", line 587, in run_until_complete
    return future.result()
  File "test.py", line 31, in main
    await page.wait_for_load_state('networkidle')
  File "C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\async_api\_generated.py", line 5903, in wait_for_load_state
    self._impl_obj.wait_for_load_state(state=state, timeout=timeout),
  File "C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\_impl\_page.py", line 478, in wait_for_load_state
    return await self._main_frame.wait_for_load_state(**locals_to_params(locals()))
  File "C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\_impl\_frame.py", line 197, in wait_for_load_state
    await wait_helper.result()
playwright._impl._api_types.TimeoutError: Timeout 30000.0ms exceeded.
Task exception was never retrieved
future: <Task finished coro=<WaitHelper.reject_on_timeout.<locals>.reject() done, defined at C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\_impl\_wait_helper.py:56> exception=CancelledError()>
concurrent.futures._base.CancelledError
```

Does webkit not support the `networkidle` state? If not, that should be a runtime error, not be allowed to go through =/

**Comment by Pomax** [comment]
If I change this line to `await page.wait_for_load_state('domcontentloaded')` things work find, so it looks like however webkit's being checked for network traffic actually breaks playwright.

**Comment by mxschmitt** [comment]
When I execute your example I get the following error message:

> Cannot parse arguments: Unknown option --window-size=1200,800

This is because the WebKit browser does not know this argument. So either specify this argument only in the browser which allow this CLI argument or only rely on the context options viewport option, which we would recommend.

**Comment by Pomax** [comment]
I see, you don't pass flags through like Puppeteer? 

I've updated it to use set_viewport_size and that indeed no longer shows a gap wher the scrollbar goes.

**Comment by Pomax** [comment]
That doesn't fix the problem though:

```
async def main():
    async with async_playwright() as p:
        for browser_type in [p.chromium, p.firefox, p.webkit]:
            browser = await browser_type.launch(headless=True)

            print(f'Navigating to {args.url} using {browser_type.name}')
            page = await browser.new_page()
            await page.set_viewport_size({ 'width': args.width, 'height': 800 })
            await page.goto(args.url)

            # this stalls to death for webkit
            await page.wait_for_load_state('networkidle')

            print(f'Creating screenshot-{args.width} for {browser_type.name}')
            await page.screenshot(path=f'example-{browser_type.name}-{args.width}.png', full_page=True)
            await browser.close()
            print('')

```

**Comment by mxschmitt** [comment]
I unfortunately can't reproduce that, I tried version 1.10 and the current master version, for both versions the script is running without any problems. You can also set the `networkidle` parameter when you navigate to your page, like `await page.goto(url, wait_until="networkidle")` but that should result in the same internally.

Are you using the latest version? 1.10.0

**Comment by Pomax** [comment]
Yep. Did a pip freeze > requirements.txt and it says `playwright==1.10.0`

**Comment by Pomax** [comment]
here's a possible reason:
```
> playwright ff https://stackoverflow.com/questions
```
works fine.

```
> playwright cr https://stackoverflow.com/questions
```

also works fine

```
> playwright wk https://stackoverflow.com/questions
```

throw up a windows firewall prompt because while FF and Chrome are normal, installed applications that get used daily, that is not the case for webkit. 

After clearing webkit, `playwright wk https://stackoverflow.com/questions` works fine, but the script shown above still times out when webkit is told to wait for `networkidle`.

**Comment by Pomax** [comment]
Run log:
```
Navigating to https://stackoverflow.com/questions using webkit
Traceback (most recent call last):
  File "test.py", line 40, in <module>
    asyncio.run(main())
  File "c:\program files\python37\lib\asyncio\runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "c:\program files\python37\lib\asyncio\base_events.py", line 587, in run_until_complete
    return future.result()
  File "test.py", line 33, in main
    await page.wait_for_load_state(wait_type)
  File "C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\async_api\_generated.py", line 5903, in wait_for_load_state
    self._impl_obj.wait_for_load_state(state=state, timeout=timeout),
  File "C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\_impl\_page.py", line 478, in wait_for_load_state
    return await self._main_frame.wait_for_load_state(**locals_to_params(locals()))
  File "C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\_impl\_frame.py", line 197, in wait_for_load_state
    await wait_helper.result()
playwright._impl._api_types.TimeoutError: Timeout 30000.0ms exceeded.
Task exception was never retrieved
future: <Task finished coro=<WaitHelper.reject_on_timeout.<locals>.reject() done, defined at C:\Users\Mike\Documents\Git\projects\ci-image-diff\venv\lib\site-packages\playwright\_impl\_wait_helper.py:56> exception=CancelledError()>
concurrent.futures._base.CancelledError
```

**Comment by Pomax** [comment]
windows firewall is never getting disabled. However, you are closed to windows than you think: you can get free Win10 VMs from Microsoft themselves for testing. 

Windows firewall will trigger a "an application is trying to access the network" on a first time executable process network attempt - think of it as Little Snitch, except only once.

**Comment by Pomax** [comment]
> What I actually meant is that I'm not well-versed in windows world.

Yep, yhat's fair. 

> Does the clearing work on a per-process basis? 

It's based on process, so any "new process" that attemps LAN or WAN access gets flagged on their first try. However, it feels like that's not the real problem here - I've been working on a visual regression tool for web development (using Playwright as the screenshooting server, and opencv/scikit for image diffing that _isn't_ based on pixels but is based on what people actually care about) and keep running into Webkit not understanding `networkidle`

I'll see if I can give you a VM-reproducible example of this happening. On my setup, it's super consistent.

**Comment by mxschmitt** [comment]
Closing since the issue seemed stale as part of the triage process. Please create a new issue if the issue is still persistent.

Having a reproducible would be awesome for that.

**Comment by jolly-xw** [comment]
I encountered the same issue when using await page.wait_for_load_state("networkidle").

Run log:
Task exception was never retrieved
future: <Task finished coro=<WaitHelper.reject_on_timeout.<locals>.reject() done, defined at D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_wait_helper.py:83> exception=CancelledError() created at D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_wait_helper.py:87>        
source_traceback: Object created at (most recent call last):
  File "d:/Project/ScreenShots/main.py", line 161, in <module>
    asyncio.run(main(), debug=True)
  File "D:\MINICONDA\envs\playwright\lib\asyncio\runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "D:\MINICONDA\envs\playwright\lib\asyncio\base_events.py", line 574, in run_until_complete
    self.run_forever()
  File "D:\MINICONDA\envs\playwright\lib\asyncio\base_events.py", line 541, in run_forever
    self._run_once()
  File "D:\MINICONDA\envs\playwright\lib\asyncio\base_events.py", line 1778, in _run_once
    handle._run()
  File "D:\MINICONDA\envs\playwright\lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "d:/Project/ScreenShots/main.py", line 46, in screen_shot_single
    await page.wait_for_load_state("networkidle")
  File "D:\MINICONDA\envs\playwright\lib\site-packages\playwright\async_api\_generated.py", line 9348, in wait_for_load_state
    await self._impl_obj.wait_for_load_state(state=state, timeout=timeout)
  File "D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_page.py", line 495, in wait_for_load_state
    return await self._main_frame.wait_for_load_state(**locals_to_params(locals()))
  File "D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_frame.py", line 242, in wait_for_load_state
    return await self._wait_for_load_state_impl(state, timeout)
  File "D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_frame.py", line 253, in _wait_for_load_state_impl
    wait_helper = self._setup_navigation_wait_helper("wait_for_load_state", timeout)
  File "D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_frame.py", line 169, in _setup_navigation_wait_helper
    wait_helper.reject_on_timeout(timeout, f"Timeout {timeout}ms exceeded.")
  File "D:\MINICONDA\envs\playwright\lib\site-packages\playwright\_impl\_wait_helper.py", line 87, in reject_on_timeout
    self._pending_tasks.append(self._loop.create_task(reject()))
concurrent.futures._base.CancelledError

**Comment by Pomax** [comment]
you probably want to open a _new_ issue and mention this one, instead of commenting on a closed issue.
